### DISCUSSION TITLE
How to Integrate AMP with Nuxt 3 (CSR Mode) - Issues with AMP Script Injection

### DISCUSSION THREAD

**Discussion by SheikhElMoctarG** [nuxt/nuxt#28358] [Questions]
Hi Nuxt Team,

I am working on a Nuxt 3 project and trying to integrate AMP support. I followed the approach outlined in [this Nuxt 2 example](https://github.com/nuxt/nuxt/blob/2.x/examples/with-amp/nuxt.config.js) for configuring AMP, but I am facing issues getting the AMP script properly injected into the <head> of my pages.

- Issue:
Despite configuring as shown, the AMP script (<script async src="https://cdn.ampproject.org/v0.js"></script>) is not appearing in the <head> of my pages. I have confirmed that the app.html template is being used, but the AMP modifications are not being applied as expected.

- Additional Context:

I am using Client-Side Rendering (CSR) mode, not Server-Side Rendering (SSR).
I am looking for guidance on correctly integrating AMP into a Nuxt 3 project where CSR is enabled.

what is the solution ?
![image](https://github.com/user-attachments/assets/3442dc67-6c37-4924-b1c2-91a4d08fcdc4)

**Comment by xjccc** [comment]
Use this hook? `render:html`

[render:html](https://nuxt.com/docs/api/advanced/hooks#nitro-app-hooks-runtime-server-side)

**Reply by SheikhElMoctarG** [reply]
yes i used but no result

**Comment by TheAlexLichter** [comment]
I doubt AMP will work without SSR or prerendering though
